src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -581,16 +597,14 @@ private static unsafe int Send(int fd, int flags, byte[] buffer, ref int offset,
                 }
             }
 
-            if (sent == -1)
+            if (errno != Interop.Error.SUCCESS)
             {
-                errno = Interop.Sys.GetLastError();
                 return -1;
             }
 
-            errno = Interop.Error.SUCCESS;
-            offset += sent;
-            count -= sent;
-            return sent;
+            offset += (int)sent;
```

Comment:
Maybe only cast sent to an int once.

Also, can you add either an assert or a "checked" cast, so no data is silently lost?